iroha/test_network/tests/mock.rs

```diff
@@ -0,0 +1,790 @@
+                    trusted_peers_ids,
+                ))
+            }
+        }
+
+        pub fn sign_tx(isi: impl IntoIterator<Item = Instruction>) -> VersionedAcceptedTransaction {
+            let tx = Transaction::new(isi.into_iter().collect(), ROOT_ID.clone(), 100_000)
+                .sign(&ROOT_KEYS)
+                .unwrap();
+            let tx = VersionedAcceptedTransaction::from_transaction(tx, 4096).unwrap();
+            dbg!(tx.hash());
+            tx
+        }
+    }
+}
+
+/// Checks if `blocks` number of blocks is applied on peer.
+async fn blocks_applied_peer(channel: &mut mpsc::Receiver<Stored>, blocks: usize) -> usize {
+    for i in 0..blocks {
+        let timeout = time::timeout(Duration::from_millis(100), channel.recv())
+            .await
+            .map(|o| o.is_none())
+            .unwrap_or(true);
+        if timeout {
+            return i;
+        }
+    }
+    blocks
+}
+
+/// Checks if `blocks` number of blocks is applied on peers.
+async fn blocks_applied(channels: &mut [mpsc::Receiver<Stored>], blocks: usize) {
+    let mut out = Vec::new();
+    for chan in channels.iter_mut() {
+        // Blocks number is increased by one in order to remove false positives,
+        // when peer actually accepted more blocks than needed.
+        out.push(blocks_applied_peer(chan, blocks + 1).await);
+    }
+    assert_eq!(out, vec![blocks; channels.len()]);
+}
+
+async fn send_tx<W, G, Q, S, K, B>(network: &Network<W, G, Q, S, K, B>, to_leader: bool)
+where
+    W: WorldTrait,
+    G: GenesisNetworkTrait,
+    Q: QueueTrait<World = W>,
+    S: SumeragiTrait<Queue = Q, GenesisNetwork = G, World = W> + Handler<sumeragi::Round>,
+    K: KuraTrait<World = W>,
+    B: BlockSynchronizerTrait<Sumeragi = S, World = W>,
+{
+    let tx = world::sign_tx(vec![]);
+    let leader = network.send(|iroha| &iroha.sumeragi, IsLeader).await;
+    let (_, peer) = leader
+        .into_iter()
+        .zip(network.peers())
+        .find(|(leader, _)| if to_leader { *leader } else { !*leader })
+        .unwrap();
+    peer.iroha
+        .as_ref()
+        .unwrap()
+        .sumeragi
+        .do_send(sumeragi::Round(vec![tx]))
+        .await;
+}
+
+async fn get_topology_with_shift<W, G, Q, S, K, B>(
+    network: &Network<W, G, Q, S, K, B>,
+    shift: u32,
+) -> InitializedNetworkTopology
+where
+    W: WorldTrait,
+    G: GenesisNetworkTrait,
+    Q: QueueTrait<World = W>,
+    S: SumeragiTrait<Queue = Q, GenesisNetwork = G, World = W> + Handler<sumeragi::NetworkTopology>,
+    K: KuraTrait<World = W>,
+    B: BlockSynchronizerTrait<Sumeragi = S, World = W>,
+{
+    let mut expected = network
+        .genesis
+        .iroha
+        .as_ref()
+        .unwrap()
+        .sumeragi
+        .send(sumeragi::NetworkTopology)
+        .await
+        .unwrap();
+    expected.shift_peers_by_n(shift);
+    expected
+}
+
+#[tokio::test(flavor = "multi_thread")]
+#[ignore = "mock"]
+async fn all_peers_commit_block() {
```

Comment:
This and below are mostly consensus tests, you can move test utils here, but the tests should stay in sumeragi.

Reply:
Lets proceed with how it is right now. After rework of sumeragi we can move it to sumeragi.

Reply:
Why do you think it is dependent on sumeragi refactoring?

Reply:
Because of usage of test_network